third_party/move/move-compiler-v2/tests/ability-check/v1-typing/pack_constraint_not_satisfied.exp

```diff
@@ -0,0 +1,45 @@
+
+Diagnostics:
+error: type `Coin` is missing required ability `drop` (type was inferred)
+  ┌─ tests/ability-check/v1-typing/pack_constraint_not_satisfied.move:8:9
+  │
+4 │     struct S<T: drop> has drop { c: T }
+  │              - declaration of type parameter `T`
+  ·
+8 │         S { c: Coin {} };
+  │         ^
+  │
+  = required by instantiating type parameter `T:drop` of struct `S`
+
+error: type `R<key>` is missing required ability `key` (type was inferred)
+   ┌─ tests/ability-check/v1-typing/pack_constraint_not_satisfied.move:12:9
+   │
+ 3 │     struct R<T: key>  { r: T }
+   │              - declaration of type parameter `T`
+   ·
+12 │         R {r: R { r: _ } } = R { r: R { r: 0 }};
+   │         ^
+   │
+   = required by instantiating type parameter `T:key` of struct `R`
```

Comment:
V1 does not have this error and the following ones, but it seems valid. 

The error message is a bit confusing, as `R` is both the outer and inner type, but I suppose it's ok.  It would be useful if we could point to `R` and say that it needs to declare `has key`, which I've mentioned on a different `.exp` file, but perhaps you should add this case to the issue filed in that case to double-check how it looks for this error.